cmd/build/v2/services.go

```diff
@@ -122,13 +131,13 @@ func (bc *OktetoBuilder) checkServiceToBuildDuringDeploy(service string, manifes
 	return nil
 }
 
-func (bc *OktetoBuilder) GetServicesToBuildForImage(ctx context.Context, manifest *model.Manifest, imgFinder model.ImageFromManifest) ([]string, error) {
+func (bc *OktetoBuilder) GetSvcToBuildFromRegex(manifest *model.Manifest, imgFinder model.ImageFromManifest) (string, error) {
 	img := imgFinder(manifest)
 	reg := regexp.MustCompile(`OKTETO_BUILD_(\w+)_`)
 	matches := reg.FindStringSubmatch(img)
 	foundMatches := 2
 	if len(matches) == 0 {
-		return nil, nil
+		return "", ErrimageIsNotAOktetoBuildSyntax
```

Comment:
Should this really fail? What would happen if they specify an env var they have defined in their env? I mean, if this is another kind of env var, e.g. `API_IMAGE`

Reply:
It's failing the function because we couldn't get the image from the regex but after it in the caller function we are capturing the error and managing it

Reply:
But we are not handling this error specifically in the caller. We only handle `ErrOktetBuildSyntaxImageIsNotInBuildSection`. Is that expected?

Reply:
We don't handle because if it's not the other error we should continue with the execution to check if the image is already built. I'll add a comment to make it clearer